src/unix/bsd/netbsdlike/netbsd/mod.rs

```diff
@@ -658,6 +673,11 @@ extern {
     pub fn newlocale(mask: ::c_int,
                      locale: *const ::c_char,
                      base: ::locale_t) -> ::locale_t;
+
+    // This should work, but it causes the netbsd CI build to fail with an
+    // intra-libc.a undefined reference to `wait6`.
+    //pub fn waitid(idtype: idtype_t, id: id_t, infop: *mut ::siginfo_t,
+    //              options: ::c_int) -> ::c_int;
```

Comment:
Yeah, OK.  To be clear, though, I'm pretty sure this is a bug within (possibly just the rumprun version of) NetBSD libc.  The actual error is

    /usr/local/rumprun-x86_64/lib/libc.a(waitid.o): In function `_waitid':
    /build/rumprun/src-netbsd/lib/libc/gen/waitid.c:52: undefined reference to `wait6'

I don't think there's any way to work around that from within our code.